documentation/modules/cruise-control/con-cruise-control-overview.adoc

```diff
@@ -2,54 +2,501 @@
+Cruise Control comprises four main components: 
+
+Load Monitor:: Load Monitor collects the metrics and analyzes cluster workload data.
+Analyzer:: Analyzer generates optimization proposals based on collected data and configured goals.
+Anomaly Detector:: Anomaly Detector identifies and reports irregularities in cluster behavior.
+Executor:: Executor applies approved optimization proposals to the cluster.
+
+Cruise Control also provides a REST API for client interactions, which Strimzi uses to support these features:
+
+* Generating optimization proposals from optimization goals
+* Rebalancing a Kafka cluster based on an optimization proposal
+
+NOTE: Other Cruise Control features are not currently supported, including self healing, notifications, and write-your-own goals.
+
+== Optimization goals
+
+Optimization goals define objectives for rebalancing, such as distributing topic replicas evenly across brokers. 
+
+They are categorized as follows:
+
+* *Hard goals* are preset and mandatory for a proposal to succeed.
+* *Soft goals* are optional and can be set aside if hard goals are met.
+* *Main goals* are inherited from Cruise Control, some preset as hard goals, used by default.
+* *Default goals* are the same as main goals by default but customizable.
+* *User-provided goals* are a subset of default goals configured for specific proposals.
+
+Configure optimization goals in the `Kafka` and `KafkaRebalance` custom resources.
+
+* `Kafka` resource for default, hard, and main goals.
+** Default goals: `Kafka.spec.cruiseControl.config.default.goals`
+** Hard goals: `Kafka.spec.cruiseControl.config.hard.goals`
+** Main goals: `Kafka.spec.cruiseControl.config.goals`
+* `KafkaRebalance` resource for user-provided goals.
+** User-Provided goals: `KafkaRebalance.spec.goals`
+
+=== Hard and soft goals
+
+Hard goals are mandatory and must be satisfied for optimization proposals to be valid. 
+Soft goals are best-effort objectives that Cruise Control tries to meet after all hard goals are satisfied.
+The classification of hard and soft goals is fixed in Cruise Control code and cannot be changed.
+
+A proposal meeting all hard goals is valid, even if it violates some soft goals. 
+Cruise Control prioritizes satisfying hard goals and then maximizes soft goals.
+
+For example, a soft goal might be to evenly distribute a topic's replicas. 
+Cruise Control will ignore this if it conflicts with hard goals.
+
+Configure hard goals in your Cruise Control deployment using `Kafka.spec.cruiseControl.config.hard.goals`:
+
+* To enforce all hard goals, omit the `hard.goals` property.
+* To specify hard goals, list them in `hard.goals`.
+* To exclude a hard goal, ensure it's not in either `default.goals` or `hard.goals`.
+
+Increasing the number of configured hard goals will reduce the likelihood of Cruise Control generating valid optimization proposals.
```

Comment:
It'll reduce the likelihood of generating an optimization proposal period. From what I understand there is no concept of an invalid optimization proposal. If the hard goals cannot be satisfied, a proposal will not be generated
```suggestion
Increasing the number of configured hard goals will reduce the likelihood of Cruise Control generating optimization proposals.
```